packages/api/internal/handlers/sandbox_resume.go

```diff
@@ -58,17 +58,31 @@ func (a *APIStore) PostSandboxesSandboxIDResume(c *gin.Context, sandboxID api.Sa
 
 	// This is also checked during in orchestrator.CreateSandbox, where the sandbox ID is reserved,
 	// but we want to do a quick check here to return an error quickly if possible.
-	sbxCache, err := a.orchestrator.GetSandbox(sandboxID)
+	sbxCache, err := a.orchestrator.GetSandbox(sandboxID, true)
 	if err == nil {
-		zap.L().Debug("Sandbox is already running",
-			logger.WithSandboxID(sandboxID),
-			zap.Time("end_time", sbxCache.GetEndTime()),
-			zap.Time("start_time", sbxCache.StartTime),
-			zap.String("node_id", sbxCache.NodeID),
-		)
-		a.sendAPIStoreError(c, http.StatusConflict, fmt.Sprintf("Sandbox %s is already running", sandboxID))
+		state := sbxCache.GetState()
+		switch state {
+		case instance.StatePaused, instance.StatePausing:
+			err = sbxCache.WaitForStop(ctx)
+			if err != nil {
+				a.sendAPIStoreError(c, http.StatusInternalServerError, "Error when resuming sandbox")
+				return
+			}
+		case instance.StateShuttingDown, instance.StateKilled:
+			a.sendAPIStoreError(c, http.StatusBadRequest, fmt.Sprintf("Sandbox %s is already killed", sandboxID))
+			return
+		case instance.StateRunning:
+			a.sendAPIStoreError(c, http.StatusConflict, fmt.Sprintf("Sandbox %s is already running", sandboxID))
```

Comment:
I think we could return success state, small issue is we are returning `201`, which would be little bit weird? Not sure about adding /changing to`200` as it could be breaking change. In our SDK, it's handled and it doesn't require 201, so both option are fine